modules/basicfuncs/list-funcs.c

```diff
@@ -329,25 +329,223 @@ tf_implode(LogMessage *msg, gint argc, GString *argv[], GString *result)
+    { NULL }
+  };
+
+  ctx = g_option_context_new((*argv)[0]);
+  g_option_context_add_main_entries(ctx, list_search_options, NULL);
+
+  if (!g_option_context_parse(ctx, argc, argv, error))
+    {
+      goto exit;
+    }
+
+  if (!_list_search_mode_str_to_string_match_mode(mode_str, mode))
+    {
+      g_set_error(error, LOG_TEMPLATE_ERROR, LOG_TEMPLATE_ERROR_COMPILE,
+                  "$(list-search) Invalid list-search mode: %s. "
+                  "Valid modes are: literal", mode_str);
+      goto exit;
+    }
+
+  result = TRUE;
+
+exit:
+  g_free(mode_str);
+  g_option_context_free(ctx);
+
+  return result;
+}
+
+static GSList *
+_char_p_array_to_str_slist(gint argc, gchar *argv[])
+{
+  GSList *slist = NULL;
+
+  for (gint i = 0; i < argc; i++)
+    slist = g_slist_append(slist, g_string_new(argv[i]));
+
+  return slist;
+}
+
+static gboolean _list_search_init_matcher(ListSearchState *state, StringMatchMode mode, gint argc, gchar *argv[],
+                                          GError **error)
+{
+  if (argc < 2)
+    {
+      g_set_error(error, LOG_TEMPLATE_ERROR, LOG_TEMPLATE_ERROR_COMPILE,
+                  "$(list-search) Pattern is missing. Usage: $(list-search [options] <pattern> ${list})");
+      return FALSE;
+    }
+  else if (argc < 3)
+    {
+      g_set_error(error, LOG_TEMPLATE_ERROR, LOG_TEMPLATE_ERROR_COMPILE,
+                  "$(list-search) List is missing. Usage: $(list-search [options] <pattern> ${list}");
+      return FALSE;
+    }
+
+  gchar *pattern = argv[1];
+  state->matcher = string_matcher_new(mode, pattern);
+
+  if (!string_matcher_prepare(state->matcher))
+    {
+      g_set_error(error, LOG_TEMPLATE_ERROR, LOG_TEMPLATE_ERROR_COMPILE,
+                  "$(list-search) Failed to prepare pattern: %s", pattern);
+      return FALSE;
+    }
+
+  state->args = _char_p_array_to_str_slist(argc-2, argv+2);
+
+  return TRUE;
 }
 
 static gboolean
 tf_list_search_prepare(LogTemplateFunction *self, gpointer s, LogTemplate *parent, gint argc, gchar *argv[],
                        GError **error)
 {
+  ListSearchState *state = (ListSearchState *)s;
+  StringMatchMode mode;
+
+  if (!_list_search_parse_options(&mode, &state->start_index, &argc, &argv, error))
+    return FALSE;
+
+  if (!_list_search_init_matcher(state, mode, argc, argv, error))
+    return FALSE;
+
   return TRUE;
 }
 
+static void _load_arg(gpointer data, gpointer user_data)
+{
+  GString *arg = (GString *)data;
+  ListScanner *scanner = (ListScanner *)user_data;
+
+  list_scanner_input_string(scanner, arg->str, arg->len);
+}
+
 static void
 tf_list_search_call(LogTemplateFunction *self, gpointer s, const LogTemplateInvokeArgs *args, GString *result)
 {
+  ListSearchState *state = (ListSearchState *)s;
+  ListScanner scanner;
+  gint index = state->start_index;
+
+  list_scanner_init(&scanner);
+  g_slist_foreach(state->args, _load_arg, &scanner);
+  list_scanner_skip_n(&scanner, index);
```

Comment:
These are not necessary, and unfortunately not working.

Not necessary to load arguments as using tf simple functions prior to call, you will have evaluated template results in `args`.

Use `args->argv` as input to `list_scanner_input_string`. 
I guess only 1 list is supported as input, so a simple call to `list_scanner_input_string` would be enough and no need for `_load_args()`

Reply:
Thanks, fixed :)